src/QsCompiler/Transformations/ClassicallyControlledTransformation.cs

```diff
@@ -4,615 +4,472 @@
+
+                            controlArgs = Utilities.CreateValueTupleExpression(conditionExpression, zeroOpArg, oneOpArg);
+
+                            targetArgs = ImmutableArray.Create(condArgs.ResolvedType, defaultArgs.ResolvedType);
                         }
-                        else if (ctl)
+                        else if (defaultScope == null)
                         {
-                            controlOpInfo = (result == QsResult.Zero)
-                            ? BuiltIn.ApplyIfZeroC
-                            : BuiltIn.ApplyIfOneC;
+                            if (adj && ctl)
+                            {
+                                controlOpInfo = (result == QsResult.Zero)
+                                ? BuiltIn.ApplyIfZeroCA
+                                : BuiltIn.ApplyIfOneCA;
+                            }
+                            else if (adj)
+                            {
+                                controlOpInfo = (result == QsResult.Zero)
+                                ? BuiltIn.ApplyIfZeroA
+                                : BuiltIn.ApplyIfOneA;
+                            }
+                            else if (ctl)
+                            {
+                                controlOpInfo = (result == QsResult.Zero)
+                                ? BuiltIn.ApplyIfZeroC
+                                : BuiltIn.ApplyIfOneC;
+                            }
+                            else
+                            {
+                                controlOpInfo = (result == QsResult.Zero)
+                                ? BuiltIn.ApplyIfZero
+                                : BuiltIn.ApplyIfOne;
+                            }
+
+                            controlArgs = Utilities.CreateValueTupleExpression(
+                                conditionExpression,
+                                Utilities.CreateValueTupleExpression(condId, condArgs));
+
+                            targetArgs = ImmutableArray.Create(condArgs.ResolvedType);
                         }
                         else
                         {
-                            controlOpInfo = (result == QsResult.Zero)
-                            ? BuiltIn.ApplyIfZero
-                            : BuiltIn.ApplyIfOne;
+                            return null; // ToDo: Diagnostic message - default body exists, but is not valid
                         }
 
-                        controlArgs = CreateValueTupleExpression(
-                            conditionExpression,
-                            CreateValueTupleExpression(condId, condArgs));
-
-                        targetArgs = ImmutableArray.Create(condArgs.ResolvedType);
                     }
                     else
                     {
-                        return null; // ToDo: Diagnostic message - default body exists, but is not valid
+                        return null; // ToDo: Diagnostic message - cond body not valid
```

Comment:
What does that mean? Also, I assume this is talking about the conditional block?

Reply:
This is referring to the body of the conditional block. Or in other words, the body that is conditional.